### ISSUE TITLE
modeline $schema does not take precedence

### ISSUE THREAD

**Issue by Kosta-Github** [redhat-developer/yaml-language-server#640]
## Describe the bug

When specifying an explicit schema via the modeline `yaml-language-server` comment
```yaml
---
# yaml-language-server: $schema=https://some.com/not-existing.yaml
openapi: '3.0.1'
```
the given schema is ignored if a matching built-in `openapi` schema is found:

![image](https://user-images.githubusercontent.com/2526664/149094650-2baf6b5f-97f2-428a-b9f5-97d04b4aacde.png)

If no matching `openapi` spec could be found (e.g., for an `openapi` version `4.x`), the explicitly defined schema is used:

![image](https://user-images.githubusercontent.com/2526664/149094585-c709834c-981a-45a8-9a13-55b14e7d33ff.png)

Note, that the behaviour is the same if the specified schema used for testing purpose actually exists.

## Expected Behavior

I would expect that the schema explicitly specified via the inline modeline `yaml-language-server` comment will be used.

## Current Behavior

See error description above.

## Steps to Reproduce

1. Use this test file
  ```yaml
  ---
  # yaml-language-server: $schema=https://some.com/not-existing.yaml
  openapi: '3.0.1'
  ```
2. Check that the given schema is used for validation.

## Environment
- [ ] Windows
- [ ] Mac
- [ ] Linux
- [ ] other (please specify)

**Comment by msivasubramaniaan** [comment]
@Kosta-Github I have added the openapi property on one of the local schema which referred and the mentioned issue was resolved in 1.4.0 Please refer the screenshot 
![image](https://user-images.githubusercontent.com/93245779/153852196-616db7b8-9220-46d5-83f7-71d22848cc77.png)

**Comment by msivasubramaniaan** [comment]
@Kosta-Github  Please confirm this issue in 1.4.0 version. If it was resolved then please close the issue

**Comment by msivasubramaniaan** [comment]
Closing the issue as it resolved and we verified in the version 1.4.0.
